zio-json/shared/src/main/scala/zio/json/decoder.scala

```diff
@@ -217,6 +217,23 @@ object JsonDecoder extends GeneratedTupleDecoders with DecoderLowPriority0 {
   implicit val bigDecimal: JsonDecoder[java.math.BigDecimal] = number(
     Lexer.bigDecimal
   )
+  implicit val dayOfWeek: JsonDecoder[java.time.DayOfWeek]           = string.map(java.time.DayOfWeek.valueOf(_))
```

Comment:
Your tests seem to cover the happy path, it would be good to see what happens if some decoding fails.

I’m mentioning this because you’re using `map` with impure functions that might throw. 
Maybe what you want is `mapOrFail`

Reply:
Yeah that's a great call on that!